datastore.cpp

```diff
@@ -229,7 +229,7 @@ int main(const int argc, const char *argv[])
         // Allocate a memory layout in shared memory, deallocate previous
         SharedMemory *layout_memory =
             SharedMemoryFactory::Get(layout_region, sizeof(SharedDataLayout));
-        SharedDataLayout *shared_layout_ptr = new (layout_memory->Ptr()) SharedDataLayout();
```

Comment:
Nope, `auto` type deduction works like template type deduction. And `T` vs `T*` matters only for human readability. Here's an example with a nice trick to force the compiler to show me the infered types. In both cases they are `char*`: http://coliru.stacked-crooked.com/a/302af5710763fb14